src/main.rs

```diff
@@ -157,3 +226,53 @@ fn parse_line(line: &str) -> anyhow::Result<Option<DVec3>> {
         _ => Ok(None),
     }
 }
+
+fn histogram(positions: &Vec<DVec3>, r_variable: fn(&DVec3) -> f64) -> (Vec<Vec<DVec3>>, Vec<f64>) {
+    let mut position_set: Vec<DVec3> = positions.clone();
+
+    let n = position_set.len();
+    let avg = position_set.iter().copied().sum::<DVec3>() / n as f64;
+    let std_dev = (position_set
+        .iter()
+        .copied()
+        .map(|r| (r - avg).powf(2.))
+        .sum::<DVec3>()
+        / (n - 1) as f64)
+        .powf(0.5);
+
+    let cutoff: i32 = 3; // measured in standard deviations
+    let div: i32 = 6;
```

Comment:
Any reason why these couldn't be floats?

Reply:
I've created **range** (iterator) through a Range<i32>. Range<f64> can't exist because of unpredictable behavior. If there's another way of creating the range variable that doesn't rely on Range<T> let me know.

Reply:
Ah right, didn't see the range, my bad.